File: 151004/Bosko/lab1/lab1/DTO/NewsResponseTo.cs
using lab1.DTO.Interface;
using lab1.Models;

namespace lab1.DTO
{
    public class NewsResponseTo(int Id, int? UserId, string? Title, string? Content, string? Created, string? Modified, User? User, ICollection<Note> Notes) : IResponseTo
    {
        public int Id { get; set; } = Id;

        public int? UserId { get; set; } = UserId;

        public string? Title { get; set; } = Title;

        public string? Content { get; set; } = Content;

        public string? Created { get; set; } = Created;

        public string? Modified { get; set;} = Modified;

        public User? User { get; set; } = User;

        public ICollection<Note> Notes { get; set; } = Notes;
    }
}

File: 151002/Strukov/REST News/src/main/java/by/bsuir/news/dto/response/NewsResponseTo.java
package by.bsuir.news.dto.response;

import by.bsuir.news.entity.News;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;

import java.sql.Timestamp;
import java.util.Date;

public class NewsResponseTo implements IResponseTO<News, NewsResponseTo> {
    private Long id;
    private String title;
    private String content;
    private Long editorId;
    private Date created;
    private Date modified;

    public NewsResponseTo toModel(News source) {
        return toResponse(source);
    }

    public static NewsResponseTo toResponse(News source) {
        NewsResponseTo model = new NewsResponseTo();
        model.id = source.getId();
        model.title = source.getTitle();
        model.content = source.getContent();
        model.created = source.getCreated();
        model.modified = source.getModified();
        model.editorId = source.getEditorId().getId();
        return model;
    }

    public NewsResponseTo() {

    }

    public void setCreated(Date created) {
        this.created = created;
    }

    public void setModified(Date modified) {
        this.modified = modified;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getEditorId() {
        return editorId;
    }

    public void setEditorId(Long editorId) {
        this.editorId = editorId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Date getCreated() {
        return created;
    }

    public Date getModified() {
        return modified;
    }
}
